recipes/security_realm_file.rb

```diff
@@ -0,0 +1,38 @@
+#
+# Cookbook Name:: cdap
+# Recipe:: security_realm_file
+#
+# Copyright © 2013-2015 Cask Data, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Manage Authentication realmfile
+realmfile = node['cdap']['cdap_site']['security.authentication.basic.realmfile']
```

Comment:
This will fail if this setting isn't configured and someone calls this recipe directly. I would put the conditional check for the attributes in this file and call this recipe always from security. It'll just no-op if it's not configured.